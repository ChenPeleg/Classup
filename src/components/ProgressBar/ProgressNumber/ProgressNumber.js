import React, { Component } from "react";
import "./ProgressNumber.scss";

class ProgressNumber extends Component {
  constructor(props) {
    super(props);
    this.currentNumberRef = React.createRef();
  }
  shouldComponentUpdate(nextProps, nextState) {
    return nextProps.current === this.props.current ? false : true;
  }

  updateLineLength(prevProps, prevState) {
    if (
      this.props.number === this.props.current &&
      this.currentNumberRef.current
    ) {
      this.props.lineWidthHandler(this.currentNumberRef.current.offsetLeft);
    }
  }
  componentDidUpdate(prevProps, prevState) {
    return this.updateLineLength(prevProps, prevState);
  }
  componentDidMount(prevProps, prevState) {
    return this.updateLineLength(prevProps, prevState);
  }

  render() {
    return (
      <div ref={this.currentNumberRef} className={this.props.className}>
        {this.props.text}
      </div>
    );
  }
}

export default ProgressNumber;
