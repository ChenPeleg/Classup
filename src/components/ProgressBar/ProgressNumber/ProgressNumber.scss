.prog_q_icon {
  display: inline;
  width: 15px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 50%;
  border: 0px solid black;
  z-index: 10;
  transition: all 0.4s;
  transition-delay: 0.3s;
}
.unanswered {
  color: grey;
}
.answered {
  color: blue;
  background-color: rgb(241, 218, 143);
}
.current_Q_num {
  font-weight: bold;
  background-color: rgb(180, 173, 151);

  color: black;
}
