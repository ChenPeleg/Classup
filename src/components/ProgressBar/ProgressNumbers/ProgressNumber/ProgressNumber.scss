@import "../../../../assets/styles/base";
.prog_q_icon {
  display: inline;
  width: 15px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 50%;
  border: 0px solid black;
  z-index: 10;
  transition: $Transition_Duration;
  @include tablet {
    font-size: $Base_fontSize;
    
  }
  @include mobile {
    padding-left: 5px;
  padding-right: 5px;
  
  }
}
.unanswered {
  color: $ProgBar_unanswered_color;
}
.answered {
  color: $ProgBar_answered_color;
  background-color: $ProgBar_answered_BGcolor;
  cursor: pointer;
}
.current_Q_num {
  transition-delay: 0.3s;
  font-weight: bold;
  @include current_question_icon;
  color: $ProgBar_current_color;
  cursor: pointer;
}
.next_ questionNum {
  @include next_question_icon;
  color: $ProgBar_current_color;
  cursor: pointer;
}
.infoNumber {
  border-radius: 0%;
}
