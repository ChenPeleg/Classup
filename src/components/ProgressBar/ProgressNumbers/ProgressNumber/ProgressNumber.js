import React, { Component } from "react";
import "./ProgressNumber.scss";

class ProgressNumber extends Component {
  constructor(props) {
    super(props);
    this.currentNumberRef = React.createRef();
  }
  shouldComponentUpdate(nextProps, nextState) {
    return nextProps.curren_qNumber === this.props.curren_qNumber
      ? false
      : true;
  }

  updateLineLength(prevProps, prevState) {
    if (
      +this.props.number + 1 === +this.props.nextUnansweredQ &&
      this.currentNumberRef.current
    ) {
      this.props.lineWidthHandler(
        this.currentNumberRef.current.getBoundingClientRect().left
      );
    }
  }
  componentDidUpdate(prevProps, prevState) {
    return this.updateLineLength(prevProps, prevState);
  }
  componentDidMount(prevProps, prevState) {
    return this.updateLineLength(prevProps, prevState);
  }

  render() {
    return (
      <div
        ref={this.currentNumberRef}
        className={this.props.className}
        onClick={(event) =>
          this.props.viewAnotherQuestionHandler(this.props.number)
        }
      >
        {this.props.text}
      </div>
    );
  }
}

export default ProgressNumber;
