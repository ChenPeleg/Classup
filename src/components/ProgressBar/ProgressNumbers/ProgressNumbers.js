import React, { Component } from "react";
import "./ProgressNumbers.scss";
import ProgressNumber from "./ProgressNumber/ProgressNumber";

class progressNumbers extends Component {
  render() {
    const { questionNumber, q_total, questionNext, info_questions } = this.props.questionCounter;

    let q_prog_array = [];

    for (let i = 1; i <= q_total; i++) {
      const txt = i;
      const wasAnswered = questionNext > i ? true : false;
      const isCurrent = i === questionNumber ? true : false;
      const isNext = i === questionNext ? true : false;
      const isInfo = info_questions.includes(i);
      q_prog_array.push({
        txt: txt,
        wasAnswered: wasAnswered,
        isCurrent: isCurrent,
        isNext: isNext,
        isInfo: isInfo,
      });
    }

    return q_prog_array.map((q) => (
      <ProgressNumber
        key={q.txt + "progKey"}
        className={`prog_q_icon ${q.wasAnswered ? "answered" : "unanswered"} ${
          q.isCurrent ? "current_Q_num" : ""
          } ${q.isNext ? "next_ questionNum" : ""} ${q.isInfo ? " infoNumber" : ""}`}
        text={q.txt}
        curren_ questionNumber={questionNumber}
        number={q_prog_array.indexOf(q)}
        lineWidthHandler={this.props.lineWidthHandler}
        viewAnotherQuestionHandler={this.props.viewAnotherQuestionHandler}
        nextUnansweredQuestion={this.props.nextUnansweredQuestion}
      />
    ));
  }
}

export default progressNumbers;
