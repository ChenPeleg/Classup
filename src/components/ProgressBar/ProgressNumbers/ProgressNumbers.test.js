import React from "react"
import { shallow } from "enzyme"
import ProgressNumbers from "./ProgressNumbers"
import ProgressNumber from "./ProgressNumber/ProgressNumber"

const totalQuestionsuestions = 7;
const mockquestionCounter = {
    questionNumber: 3,
    q_total: totalQuestionsuestions,
    questionNext: 4,
    info_questions: [2, 1],
}
const wrap = shallow(<ProgressNumbers questionCounter={mockquestionCounter} />)


describe("<ProgressNumber>", () => {
    it("renders correctly", () => {
        expect(wrap).toMatchSnapshot()
    })
    it("renders the proper number of numbers", () => {
        expect(wrap.length).toBe(totalQuestionsuestions)
    })

})