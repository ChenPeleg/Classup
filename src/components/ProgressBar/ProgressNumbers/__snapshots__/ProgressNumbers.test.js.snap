// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressNumber> renders correctly 1`] = `
Array [
  <ProgressNumber
    className="prog_q_icon answered    infoNumber"
    curren_ questionNumber={3}
    key="1progKey"
    number={0}
    text={1}
  />,
  <ProgressNumber
    className="prog_q_icon answered    infoNumber"
    curren_ questionNumber={3}
    key="2progKey"
    number={1}
    text={2}
  />,
  <ProgressNumber
    className="prog_q_icon answered current_Q_num  "
    curren_ questionNumber={3}
    key="3progKey"
    number={2}
    text={3}
  />,
  <ProgressNumber
    className="prog_q_icon unanswered  next_ questionNum "
    curren_ questionNumber={3}
    key="4progKey"
    number={3}
    text={4}
  />,
  <ProgressNumber
    className="prog_q_icon unanswered   "
    curren_ questionNumber={3}
    key="5progKey"
    number={4}
    text={5}
  />,
  <ProgressNumber
    className="prog_q_icon unanswered   "
    curren_ questionNumber={3}
    key="6progKey"
    number={5}
    text={6}
  />,
  <ProgressNumber
    className="prog_q_icon unanswered   "
    curren_ questionNumber={3}
    key="7progKey"
    number={6}
    text={7}
  />,
]
`;
